### DISCUSSION TITLE
Xiao wiring

### DISCUSSION THREAD

**Discussion by LeducH** [manna-harbour/miryoku#359] [Q&A] [answered]
I’m working on a project involving the Seeed Studio XIAO nRF52840, and I need some advice on setting up the wiring and firmware.

 I have been referencing a Corne wiring diagram for guidance.
https://images.app.goo.gl/gLrgcmPCPgA1cunM8

Questions:

For the Miryoku firmware setup, should I define the board as seeeduino_xiao_ble and assign corne_left and corne_right as shields?
Regarding wiring, is it possible to follow the same layout frequency as in the Corne diagram? For instance, would connecting P.1.15 on the Pro Micro/niceNano pinout to a similar pin on the Seeed XIAO work? The labeling appears different, so I’m unsure where to assign each column and row of my Corne on the Seeed XIAO.

**Comment by manna-harbour** [comment, marked as answer]
Shields are specific to the controller, so you can't build corne with xiao. For handwires it's recommended to wire according to an existing shield so you don't have to write a shield definition. Look for a xiao shield that has a layout compatible with your keyboard. A superset of your layout will also work. See https://github.com/manna-harbour/miryoku_zmk#supported-keyboards for info on supported keyboards in Miryoku ZMK and https://github.com/stars/manna-harbour/lists/36-keys for a list of 3x5+3 ergo splits.
